### PR TITLE
Add LICENSE file for expectant auditors

### PR DESCRIPTION
**Pull request by sigmavirus24**
The classifier in setup.py should be enough for most auditors, but some
have automated tooling that checks for LICENSE files and reads those.
This helps all auditors (downstream redistributors and users) who can
not just rely on the classifier.

Related to #612

### CODE REVIEW

**Comment by sigmavirus24** [comment]
Ah, forgot the changelog entry.

File: CHANGES.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ v25.1.6
 -------
 
 * #725
+* #612 via #730: Add a LICENSE file which neds to be provided by the terms of
```
Comment (nicktimko) [inline CHANGES.rst]: needs?

Comment (jaraco) [inline CHANGES.rst]: Also, this entry should go in a new section for v25.2.0.

Comment (sigmavirus24) [inline CHANGES.rst]: Done & done.

**Comment by jakirkham** [comment]
Do you think there is any value to having a `.txt` extension for the license? There seem to be many authors that do this. Particularly to ensure that Windows is able to open this file correctly.

**Comment by sigmavirus24** [comment]
@jakirkham not really? Almost every project I've ever seen has had the license file sans `.txt`.

**Comment by jakirkham** [comment]
Just to provide some examples, NumPy [does]( https://github.com/numpy/numpy/blob/e1f191c46f2eebd6cb892a4bfe14d9dd43a06c4e/LICENSE.txt ). Similarly so does [SciPy]( https://github.com/scipy/scipy/blob/d2b4e1786bc4737849695c67f32fedd81ec9466b/LICENSE.txt ) and [Cython]( https://github.com/cython/cython/blob/0dc185de9842e1e05b6448ca45ef0e8083f35daa/LICENSE.txt ). If one looks around a bit, one can find many more examples. Though it is true there are examples without the extension.

**Comment by sigmavirus24** [comment]
Yeah, we can both dig up numerous examples either way both inside and outside of the Python ecosystem. I don't think it really matters and this (merged) pull request certainly isn't the place for a discussion of this type (whether or not to put .txt at the end of a LICENSE file name because some operating systems do weird things in an attempt to be *smart*).

**Pull request merged**